tests/unit/test_database.py

```diff
@@ -96,41 +105,45 @@ def test_connect(
 
     db_client._connect()
 
+    relation_database_password = str(postgresql_relation_data.get("password"))
     query = (
         "dbname='synapse' user='user' host='myhost' "
-        f"password='{datasource_postgresql_password}' connect_timeout=5"
+        f"password='{relation_database_password}' connect_timeout=5"
```

Comment:
Suggestion: If I am not mistaken, this is the only test that asserts something that is set on relation data.
I think it would be easier to set the relation data similar to this [Discourse PR](https://github.com/canonical/discourse-k8s-operator/pull/105), even by using a constant for all the tests and only changing the password in this one. WDYT?

Reply:
Good point, I have re-introduced the postgresql fixture and by default we don't set a password on relation data. We only explicitly set the password in the tests where we need it.